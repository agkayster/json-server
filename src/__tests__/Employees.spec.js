import employees from '../db';

test('the employees data is correct', () => {
	expect(employees).toMatchSnapshot();
});

test('the employees data length is correct', () => {
	expect(employees).toHaveLength(3);
});

test('each employees first name is correct', () => {
	expect(employees.map((employee) => employee.first_name)).toEqual([
		'Sebastian',
		'Steve',
		'Ann',
	]);
});
test('each employees last name is correct', () => {
	expect(employees.map((employee) => employee.last_name)).toEqual([
		'Eschweiler',
		'Palmer',
		'Smith',
	]);
});

for (let i = 0; i < employees.length; i += 1) {
	test(`employees[${i}] should have properties(id, first_name, last_name, email)`, () => {
		expect(employees[i]).toHaveProperty('id');
		expect(employees[i]).toHaveProperty('first_name');
		expect(employees[i]).toHaveProperty('last_name');
		expect(employees[i]).toHaveProperty('email');
	});
}

test('employees returns Smith last', () => {
	const employee1 = employees[0];
	const employee2 = employees[1];
	const employee3 = employees[2];

	const employee = jest.fn((currentEmployee) => currentEmployee.last_name); // you can change the properties you are checking for based on the properties defined in the object//

	employee(employee1);
	employee(employee2);
	employee(employee3);

	expect(employee).toHaveNthReturnedWith(2, 'Palmer'); // you pick the particular return call out of the above three you want to focus on//
	expect(employee).toHaveLastReturnedWith('Smith');
});

test('employee data has Sebastian Eschweiler and matches an object', () => {
	const SebastianEschweiler = {
		id: 1,
		first_name: 'Sebastian',
		last_name: 'Eschweiler',
		email: 'sebastian@codingthesmartway.com',
	};

	expect(employees[0]).toMatchObject(SebastianEschweiler);
});
