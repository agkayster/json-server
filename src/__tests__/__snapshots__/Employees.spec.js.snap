// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the employees data is correct 1`] = `
Array [
  Object {
    "email": "sebastian@codingthesmartway.com",
    "first_name": "Sebastian",
    "id": 1,
    "last_name": "Eschweiler",
  },
  Object {
    "email": "steve@codingthesmartway.com",
    "first_name": "Steve",
    "id": 2,
    "last_name": "Palmer",
  },
  Object {
    "email": "ann@codingthesmartway.com",
    "first_name": "Ann",
    "id": 3,
    "last_name": "Smith",
  },
]
`;
